server/runner/lib/runner/qa/client.ex

```diff
@@ -41,7 +43,17 @@ defmodule Runner.QA.Client do
       }) do
     step_url = qa_run_url(server_url, account_handle, project_handle, run_id, "/steps/#{step_id}")
 
-    qa_server_request(:patch, step_url, auth_token, json: %{result: result, issues: issues})
+    Task.start(fn ->
+      case qa_server_request(:patch, step_url, auth_token, json: %{result: result, issues: issues}) do
+        {:ok, _} ->
+          Logger.debug("Successfully updated step #{step_id}")
+
+        {:error, reason} ->
+          Logger.error("Failed to update step #{step_id}: #{inspect(reason)}")
+      end
+    end)
+
```

Comment:
I'd consider changing the name of the method to signal it's happening in the background